### PR TITLE
More parser fixes

### PR DESCRIPTION
**Pull request by EMattfolk**
This should fix some of the more nasty error messages.

### CODE REVIEW

File: sylt-parser/src/statement.rs [hunk 1]
```diff
@@ -793,14 +793,21 @@ pub fn statement<'t>(ctx: Context<'t>) -> ParseResult<'t, Statement> {
...
 
-            match (assignment(ctx), expression(ctx)) {
-                (Ok((ctx, kind)), _) => (ctx, kind),
-                (_, Ok((ctx, value))) => (ctx, StatementExpression { value }),
-                (Err((_, mut ass_errs)), Err((_, mut expr_errs))) => {
-                    ass_errs.append(&mut expr_errs);
-                    ass_errs.push(syntax_error!(ctx, "Neither an assignment or an expression"));
-                    return Err((ctx, ass_errs));
-                }
+            // Probe if we should actually parse a assignable here.
```
Comment (FredTheDino) [inline sylt-parser/src/statement.rs:796]:
```suggestion
            // Probe if this is an assignable.
```

File: tests/bugs/missleading_blob_301.sy [hunk 2]
```diff
@@ -9,7 +9,4 @@ start :: fn do
         b: "b",
     }
 end
-// error: @7
-// error: @8
 // error: @9
-// error: @10
```
Comment (FredTheDino) [inline tests/bugs/missleading_blob_301.sy:15]: <3

**Review by FredTheDino — APPROVED** [review]
LGTM - sicko!

File: sylt-parser/src/statement.rs (hunk 1 above)
Comment (EMattfolk) [reply sylt-parser/src/statement.rs:796]:
```suggestion
            // Probe if we should parse an assignable here.
```
?

**Review by sornas — APPROVED** [review]

**Pull request merged**